### ISSUE TITLE
amp-iframe inside amp-carousel occasionally fails to load.

### ISSUE THREAD

**Issue by jpettitt** [ampproject/amphtml#7810]
On pages where there is an `amp-carousel` as the lead item with an `amp-iframe` containing video as the first slide we're occasionally seeing the iframe fail to load.  Specifically Chrome dev tools shows not attempt being made to launch the iframe.  

Navigating to the next slide and back brings up the video frame.   We're seeing this about 1 in 20 times (Chrome iPhone 6+ emulation, cache off).  It feels like a race condition so it may behave differently on different networks.

Sample page

https://sacbee.relaymedia.com/amp/news/state/california/water-and-drought/article134918039.html

**Comment by jridgewell** [comment]
Assigning to myself as well, I'm curious whether this would be fixed by Layers (likely).

**Comment by camelburrito** [comment]
@jridgewell - do you have a theory for why it might have happened (it happened to me once the first time i loaded the page)

**Comment by jridgewell** [comment]
Likely culprits:

- Using `#firstAttachedCallback`
  - It's unusual in our codebase, we usually use `#buildCallback`
  - Are attributes guaranteed in `firstAttachedCallback`?
- Carousel's reparenting, due to ownership concept.

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by aghassemi** [comment]
let's verify with carousel v2

**Comment by nainar** [comment]
Closing since the original repro URL doesn't exist anymore. Please file a new issue if needed.
